src/structures/MessageMentions.js

```diff
@@ -175,7 +175,7 @@ class MessageMentions {
   has(data, { ignoreDirect = false, ignoreRoles = false, ignoreEveryone = false } = {}) {
     if (!ignoreEveryone && this.everyone) return true;
     if (!ignoreRoles && data instanceof GuildMember) {
-      for (const role of this.roles.values()) if (data.roles.has(role.id)) return true;
+      for (const role of this.roles.cache.values()) if (data.roles.has(role.id)) return true;
```

Comment:
```suggestion
      for (const role of this.roles.cache.values()) if (data.roles.cache.has(role.id)) return true;
```
above checks if `data` is a GuildMember and `GuildMember#roles` is a datastore, so roles.cache